app/controllers/AddAdditionalMetadataController.scala

```diff
@@ -34,248 +34,104 @@ class AddAdditionalMetadataController @Inject() (
-    valuesThatWereSelectedThatHaveDependencies
   }
 
-  private def getDefaultFieldsForForm(
-      metadataType: String,
-      isMainForm: Boolean,
-      staticMetadata: Set[StaticMetadata],
-      consignmentId: UUID,
-      request: Request[AnyContent]
-  ): Future[FormData] = {
+  private def getDefaultFieldsForForm(consignmentId: UUID, request: Request[AnyContent]): Future[List[FormField]] = {
     for {
       customMetadata <- customMetadataService.getCustomMetadata(consignmentId, request.token.bearerAccessToken)
       customMetadataUtils = new CustomMetadataUtils(customMetadata)
 
-      dependencyProperties: Set[CustomMetadata] = staticMetadata.flatMap { propertyNameAndValue =>
-        getDependenciesFromValue(customMetadataUtils, Set(propertyNameAndValue.name), Some(propertyNameAndValue.value))
-      }
+      dependencyProperties: Set[CustomMetadata] = getDependenciesFromValue(customMetadataUtils, Set(closureType.name), closureType.value)
+
       formFields = customMetadataUtils.convertPropertiesToFormFields(dependencyProperties)
     } yield {
-      val pageType: String = if (isMainForm) s"Main-$metadataType" else s"Dependency-$metadataType"
-      val defaultFormData = FormData(dependencyProperties, formFields)
-      cache.set(s"$pageType-propertiesAndFieldValues", defaultFormData, 1.hour)
-      defaultFormData
-    }
-  }
-
-  private def getDependenciesFromValue(customMetadataUtils: CustomMetadataUtils, propertyNames: Set[String], valueToGetDependenciesFrom: Option[String]): Set[CustomMetadata] = {
-    propertyNames.flatMap { propertyName =>
-      valueToGetDependenciesFrom match {
-        case Some(valueToGetDependencies) =>
-          val valuesByProperties: Map[String, List[CustomMetadata.Values]] = customMetadataUtils.getValuesOfProperties(Set(propertyName))
-          val allValuesForProperty: Seq[CustomMetadata.Values] = valuesByProperties(propertyName)
-          val value: Seq[CustomMetadata.Values] = allValuesForProperty.filter(_.value.toLowerCase == valueToGetDependencies.toLowerCase)
-          val dependencyNames: Seq[String] = value.flatMap(_.dependencies.map(_.name))
-          customMetadataUtils.getCustomMetadataProperties(dependencyNames.toSet)
-        case None =>
-          customMetadataUtils.getCustomMetadataProperties(Set(propertyName))
-      }
+      cache.set("fieldValues", formFields, 1.hour)
+      formFields
     }
   }
 
-  private def getFilesFromConsignment(files: List[getConsignmentFilesMetadata.GetConsignment.Files]): List[File] = {
-    files.map(file => {
-      val filePath = file.fileMetadata.find(_.name == clientSideOriginalFilepath).map(_.value).getOrElse("")
-      File(file.fileId, filePath)
-    })
+  private def getDependenciesFromValue(customMetadataUtils: CustomMetadataUtils, propertyName: Set[String], valueToGetDependenciesFrom: String): Set[CustomMetadata] = {
```

Comment:
Is there a reason the `propertyName` parameter is a `Set[String]`? Can it not be just a `String`?

Reply:
Actually, it's an existing code but yeah, we can use `String`.